mica/archive/asp_l1.py

```diff
@@ -147,6 +150,59 @@ def get_files(obsid=None, start=None, stop=None,
                              revision=revision, content=content)
 
 
+def get_atts(obsid=None, start=None, stop=None, revision=None, filter=True):
+    """
+    Get the ground aspect solution quaternions and times covering obsid or start to stop,
+    in the ACA frame.
+
+    :obsid: obsid
+    :start: start time (DateTime compat)
+    :stop: stop time (DateTime compat)
+    :revision: aspect pipeline processing revision (integer version, None, or 'last')
+
+    :returns: Nx4 np.array of quaternions, np.array of N times, list of dict with header from each asol file.
+    """
+    if revision == 'all':
+        raise ValueError("revision 'all' doesn't really make sense for this function")
+    # These are in time order by default from get_files
+    asol_files = get_files(obsid=obsid, start=start, stop=stop,
+                           revision=revision, content=['ASPSOL'])
+    acal_files = get_files(obsid=obsid, start=start, stop=stop,
+                           revision=revision, content=['ACACAL'])
+    aqual_files = get_files(obsid=obsid, start=start, stop=stop,
+                            revision=revision, content=['ASPQUAL'])
+    # There should be one asol and one acal file for each aspect interval in the range
+    att_chunks = []
+    time_chunks = []
+    records = []
+    for asol_f, acal_f, aqual_f in zip(asol_files, acal_files, aqual_files):
+        asol = Table.read(asol_f)
+        acal = Table.read(acal_f)
+        aqual = Table.read(aqual_f)
+        # Check that the time ranges match from the fits headers (meta in the table)
+        if not np.allclose(np.array([asol.meta['TSTART'], asol.meta['TSTOP']]),
+                           np.array([acal.meta['TSTART'], acal.meta['TSTOP']]),
+                           atol=10):
+            raise ValueError("ACAL and ASOL have mismatched time ranges")
+        if filter and np.any(aqual['asp_sol_status'] != 0):
+            # For each sample with bad status find the overlapping time range in asol
+            # and remove from set
+            for idx in np.flatnonzero(aqual['asp_sol_status']):
```

Comment:
Instead of worrying about whether the aqual time covers the previous 4 asol samples or the next 4 samples, here I've just excluded all asol within 4 samples (1.025s).